docs: --pass-args (#3636)
earthly/earthly: docs/earthfile/earthfile.md

earthly --allow-privileged +my-target
```

##### `--pass-args`

> Earthly will automatically pass all existing arguments to referenced targets that exist in the _same_ Earthfile.
> When the `--pass-args` flag is set, Earthly will also propagate all arguments to the externally referenced target.

Reviewer: Improve clarity a bit.

```suggestion
Earthly automatically passes all current arguments to referenced targets in the _same_ Earthfile. However, when the `--pass-args` flag is set, Earthly will also propagate all arguments to an externally referenced target.
```
